### ISSUE TITLE
Polars module non existent attributes are inferred as Any

### ISSUE THREAD

**Issue by Hunterlige** [microsoft/pyright#10886]
**Describe the bug**
All attributes of the `polars` module are inferred as Any, including attributes that do not exist. This means that incorrect or misspelled attribute access will not be caught at type-checking time, reducing type safety.

**Code or Screenshots**

```python
import polars as pl
from typing import reveal_type


reveal_type(pl.should_not_be_any())
```

```
> aaa.py:5:13 - information: Type of "pl.should_not_be_any()" is "Any"
```

Polars version
```
 name         : polars                           
 version      : 1.31.0                           
 description  : Blazingly fast DataFrame library
```

## Expected behavior

- Accessing a non-existent attribute on the polars module should result in a type-checking error.

**Comment by LeeeeT** [comment]
Probably because of [this line](https://github.com/pola-rs/polars/blob/3c22780e7ec8035c94ab408a29ca1cd8573e744d/py-polars/polars/__init__.py#L458). I suggest wrapping the function into `if not TYPE_CHECKING:`.

**Comment by Hunterlige** [comment]
Created an issue on polars repository, closing this one. Thank you for confirming that this is not pyright related
